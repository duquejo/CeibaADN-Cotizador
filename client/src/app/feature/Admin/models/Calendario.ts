export interface Calendario {
  id?: number;
  nombre: string;
  descripcion: string;
  festivos: Array<string>;
}
