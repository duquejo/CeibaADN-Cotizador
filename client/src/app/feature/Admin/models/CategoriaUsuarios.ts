export interface CategoriaUsuarios {
  id?: number;
  nombre: string;
  descripcion: string;
  valorAlta: number;
  valorBaja: number;
}
