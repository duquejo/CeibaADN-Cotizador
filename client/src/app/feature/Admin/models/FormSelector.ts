export interface ISelectionOptions {
    label: string;
    value: string;
}
