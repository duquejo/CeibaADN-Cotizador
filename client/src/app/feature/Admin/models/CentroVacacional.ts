import { Calendario } from "./Calendario";
import { CategoriaUsuarios } from "./CategoriaUsuarios";

export interface CentroVacacional {
  id?: number;
  nombre: string;
  descripcion: string;
  calendarios: Array<Calendario>;
  categoriaUsuarios: Array<CategoriaUsuarios>;
  calendarioActivo: number;
}
