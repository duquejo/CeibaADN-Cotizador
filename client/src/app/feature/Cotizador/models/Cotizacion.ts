import { CategoriaUsuarios } from 'app/feature/Admin/models/CategoriaUsuarios';
import { CentroVacacional } from '../../Admin/models/CentroVacacional';

export interface Cotizacion {
  id?: number;
  codigo?: string;
  centroVacacional: number | CentroVacacional;
  categoriaUsuarios: number | CategoriaUsuarios;
  fechaInicio: string;
  personas: number;
  fechaFin: string;
  total?: number;
}
