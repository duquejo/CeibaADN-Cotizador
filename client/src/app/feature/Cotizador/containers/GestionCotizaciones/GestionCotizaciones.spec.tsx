import * as React from 'react';
import { ShallowWrapper, shallow } from 'enzyme';
import { GestionCotizaciones } from './index';
import { SinonStub, stub } from 'sinon';
import { CentroVacacional } from '../../../Admin/models/CentroVacacional';


describe('GestionCotizaciones Test', () => {

  let componentWrapper: ShallowWrapper;

  let componentProps: React.ComponentProps<typeof GestionCotizaciones> & {
    centrosVacacionales: CentroVacacional[];
    onSubmit: SinonStub;
    listarCentrosVacacionales:SinonStub;
    agregarNuevaCotizacion: SinonStub;
  };

  beforeEach( () => {
    componentProps = {
      onSubmit: stub(),
      listarCentrosVacacionales: stub(),
      agregarNuevaCotizacion: stub(),
      centrosVacacionales: [{
        id: 1,
        nombre: 'Centro de prueba',
        descripcion: 'Descripción de prueba',
        calendarios: [],
        categoriaUsuarios: [],
        calendarioActivo: 1
      }]
    };
    componentWrapper = shallow(<GestionCotizaciones { ...componentProps } />);
});  

  it('Debería renderizar el componente', () => {
    expect(componentWrapper).toMatchSnapshot();
  });

});
