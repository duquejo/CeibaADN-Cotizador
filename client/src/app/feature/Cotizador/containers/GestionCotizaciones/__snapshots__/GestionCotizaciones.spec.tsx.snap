// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GestionCotizaciones Test Debería renderizar el componente 1`] = `
<Fragment>
  <p>
    Proporcione la información según la disponibilidad de los calendarios que vaya seleccionando en el formulario
  </p>
  <styled.div>
    <CrearCotizacion
      centrosvacacionales={
        Array [
          Object {
            "calendarioActivo": 1,
            "calendarios": Array [],
            "categoriaUsuarios": Array [],
            "descripcion": "Descripción de prueba",
            "id": 1,
            "nombre": "Centro de prueba",
          },
        ]
      }
      onSubmit={[Function]}
    />
  </styled.div>
</Fragment>
`;
